src/utilities/views.py

```diff
@@ -1,47 +1,126 @@
+from collections import OrderedDict
+
+from django.http import Http404, JsonResponse
+from django.conf import settings
 from django.shortcuts import render
 
+from django.core import signing
+from django.core.signing import BadSignature
 from django.contrib.auth.mixins import LoginRequiredMixin
+from django.contrib.contenttypes.models import ContentType
 from django.contrib.flatpages.models import FlatPage
 from django.contrib.sites.models import Site
 from django.urls import reverse, reverse_lazy
 from django.utils.decorators import method_decorator
 from django.views.generic.edit import CreateView, UpdateView, DeleteView
 from django.views.generic.list import ListView
+from django.template.defaultfilters import capfirst
 
 from hackerspace_online.decorators import staff_member_required
 
 from .models import MenuItem, VideoResource
 from utilities.forms import MenuItemForm, VideoForm, CustomFlatpageForm
 from tenant.views import non_public_only_view, NonPublicOnlyViewMixin
 
-from dal import autocomplete
-
+from django_select2.views import AutoResponseView
+from queryset_sequence import QuerySetSequence
 
-class ModelAutocomplete(NonPublicOnlyViewMixin, autocomplete.Select2QuerySetView):
-    """ 
-    DRY autocomplete view for select2 widgets.  Example usage in urls.py with model = Quest
-    `path('quest/autocomplete/', utilities.views.ModelAutocomplete.as_view(model=Quest), name='quest_autocomplete')`
 
-    https://django-autocomplete-light.readthedocs.io/en/master/tutorial.html#create-an-autocomplete-view
+class QuerySetSequenceAutoResponseView(AutoResponseView):
     """
+    View that handles requests from heavy model widgets.
 
-    def get_queryset(self):
-        # Don't forget to filter out results depending on the visitor !
-        if not self.request.user.is_authenticated:
-            return self.model.objects.none()
-
-        # if model doesn't have a 'name' then try 'title'
-        search_field = 'name' if hasattr(self.model, 'name') else 'title'
+    The view only supports HTTP's GET method.
 
```

Comment:
This has the exact same docstring as the AutoResponseView from tags.views.  Can you add more detail specifically what this class is for?

Reply:
fixed, see latest PR from me